R/appDependencies.R

```diff
@@ -11,19 +11,43 @@
 #' # Remote installation
 #'
 #' When deployed, the app must first install all of these packages, and
-#' rsconnect does its best to ensure that all the packages have the same
-#' version as you are running locally. It knows how to install packages from
-#' the following sources:
+#' rsconnect ensures the versions used on the server will match the versions
+#' you used locally. It knows how to install packages from the following
+#' sources:
 #'
-#' * CRAN (`CRAN`) and CRAN-like repositories (`CustomCRANLikeRepository`).
-#' * BioConductor (`Bioconductor`)
-#' * Packages installed from GitHub (`github`), GitLab (`gitlab`), or
-#'   BitBucket (`bitbucket`).
+#' * CRAN and BioConductor (`Source: CRAN` or `Source: Bioconductor`). The
+#'   remote server will ignore the specific CRAN or Bioconductor mirror that
+#'   you use locally, always using the CRAN/BioC mirror that has been configured
+#'   on the server.
 #'
-#' It does not know how to install packages that you have built and installed
-#' locally so if you attempt to deploy an app that depends on such a package
-#' it will fail. To resolve this issue, you'll need to install from a known
-#' source.
+#' * Other CRAN like and CRAN-like repositories. These packages will have
```

Comment:
Thanks for capturing this; recording the current behavior will help us understand the effects of other options (e.g. preserving `Repository` from package descriptions, when available).